src/backend/utils/misc/guc_funcs.c

```diff
@@ -143,7 +143,12 @@ ExecSetVariableStmt(VariableSetStmt *stmt, bool isTopLevel)
 		case VAR_RESET:
 			(void) set_config_option(stmt->name,
 									 NULL,
-									 (superuser() ? PGC_SUSET : PGC_USERSET),
```

Comment:
Why it only affects RESET command, but not SET or RESET ALL command?

Reply:
because it's just a draft to illustrate my proposal.

I will add it to all `(superuser() ? PGC_SUSET : PGC_USERSET)`  calls,
but first I want to figure out, if this change may cause any security issues.